usr/lib/pkcs11/common/asn1.c

```diff
@@ -571,7 +573,99 @@ ber_decode_SEQUENCE( CK_BYTE  * seq,
    return CKR_FUNCTION_FAILED;
 }
 
-
+//
+//
+CK_RV
+ber_encode_CHOICE(CK_BBOOL    length_only,
+		  CK_BYTE     option,
+		  CK_BYTE  ** str,
+		  CK_ULONG  * str_len,
+		  CK_BYTE   * data,
+		  CK_ULONG    data_len)
+{
+	CK_BYTE  *buf = NULL;
+	CK_ULONG  len;
+
+	/*
+	 *  if data_len < 127 use short-form length id
+	 *  if data_len < 65536 use long-form length id with 2-byte length field
+	 */
+
+	if (data_len < 128)
+		len = 1 + 1 + data_len;
+	else if (data_len < 256)
+		len = 1 + (1 + 1) + data_len;
+	else if (data_len < (1 << 16))
+		len = 1 + (1 + 2) + data_len;
+	else if (data_len < (1 << 24))
+		len = 1 + (1 + 3) + data_len;
+	else {
```

Comment:
if one of the if statements have { }, then all the rest should also have. Please check coding-style.